### PR TITLE
Document the driver_version parameter

### PR DESCRIPTION
**Pull request by markdewing**
Addresses #3915

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Documentation changes

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
laptop

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- N/A. Code added or changed in the PR has been clang-formatted
- N/A. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- Yes. Documentation has been added (if appropriate)

### CODE REVIEW

File: docs/input_overview.rst [hunk 1]
```diff
@@ -109,10 +109,21 @@ Output file names also contain the series number, starting at the value given by
...
+~~~~~~~~~~~
+Batched drivers check against ``max_seconds`` and make efforts to stop the execution cleanly at the end of a block before reaching the maximum time. Classic drivers can also take the now-deprecated ``maxcpusecs`` parameter for the same effect in the per driver XML section.
 
 In addition, a file named ``id`` plus ``.STOP``, in this case ``vmc.STOP``, stops QMCPACK execution on the fly cleanly once being found in the working directory.
 
+
+Driver version
+~~~~~~~~~~~~~~
+The ``driver_version`` parameter selects between the new performance-portable batched drivers and the previous drivers (now referred to as the 'legacy drivers').
+The values for this parameter are ``legacy`` or ``batch`` (alternately, ``batched``).
```
Comment (prckent) [inline docs/input_overview.rst:124]: Do we actually support both batch and batched?

Comment (markdewing) [reply docs/input_overview.rst:124]: https://github.com/QMCPACK/qmcpack/blob/e99b6156f8df68c2501ed16388ecde0668e22530/src/Utilities/ProjectData.h#L54

**Review by prckent — APPROVED** [review]
Thanks for tackling this Mark.

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**